src/main/java/com/accelad/math/doubledouble/DoubleDouble.java

```diff
@@ -7,58 +7,47 @@
+    public static final DoubleDouble ZERO = fromOneDouble(0.0);
 
-    public static final DoubleDouble HALF = new DoubleDouble(0.5);
+    public static final DoubleDouble HALF = fromOneDouble(0.5);
 
-    public static final DoubleDouble PI = new DoubleDouble(3.141592653589793116e+00,
+    public static final DoubleDouble PI = fromTwoDouble(3.141592653589793116e+00,
             1.224646799147353207e-16);
 
-    public static final DoubleDouble TWO_PI = new DoubleDouble(6.283185307179586232e+00,
+    public static final DoubleDouble TWO_PI = fromTwoDouble(6.283185307179586232e+00,
             2.449293598294706414e-16);
 
-    public static final DoubleDouble PI_2 = new DoubleDouble(1.570796326794896558e+00,
+    public static final DoubleDouble PI_2 = fromTwoDouble(1.570796326794896558e+00,
             6.123233995736766036e-17);
 
-    public static final DoubleDouble E = new DoubleDouble(2.718281828459045091e+00,
+    public static final DoubleDouble E = fromTwoDouble(2.718281828459045091e+00,
             1.445646891729250158e-16);
 
-    public static final DoubleDouble NaN = new DoubleDouble(Double.NaN, Double.NaN);
+    public static final DoubleDouble NaN = fromTwoDouble(Double.NaN, Double.NaN);
 
-    public static final DoubleDouble POSITIVE_INFINITY = new DoubleDouble(Double.POSITIVE_INFINITY,
+    public static final DoubleDouble POSITIVE_INFINITY = fromTwoDouble(Double.POSITIVE_INFINITY,
             Double.POSITIVE_INFINITY);
 
-    public static final DoubleDouble NEGATIVE_INFINITY = new DoubleDouble(Double.NEGATIVE_INFINITY,
+    public static final DoubleDouble NEGATIVE_INFINITY = fromTwoDouble(Double.NEGATIVE_INFINITY,
             Double.NEGATIVE_INFINITY);
 
     public static final double EPS = 1.23259516440783e-32; /* = 2^-106 */
-
     private static final double SPLIT = 134217729.0D; // 2^27+1, for IEEE double
 
-    private static final int MAX_PRINT_DIGITS = 32;
-    private static final DoubleDouble TEN = new DoubleDouble(10.0);
+    public static final DoubleDouble ONE = fromOneDouble(1.0);
+    public static final DoubleDouble MINUS_ONE = fromOneDouble(-1.0);
+    public static final DoubleDouble TWO = fromOneDouble(2.0);
+    private static final DoubleDouble TEN = fromOneDouble(10.0);
     private static final DoubleDouble LOG_TEN = TEN.log();
-    public static final DoubleDouble ONE = new DoubleDouble(1.0);
-    public static final DoubleDouble TWO = new DoubleDouble(2.0);
-    private static final String SCI_NOT_EXPONENT_CHAR = "E";
-    private static final String SCI_NOT_ZERO = "0.0E0";
-
-    private static int magnitude(double x) {
-        final double xAbs = Math.abs(x);
-        final double xLog10 = Math.log(xAbs) / Math.log(10);
-        int xMag = (int) Math.floor(xLog10);
-
-        final double xApprox = Math.pow(10, xMag);
-        if (xApprox * 10 <= xAbs) {
-            xMag += 1;
-        }
-
-        return xMag;
-    }
+    public static final int POW_INT_CACHE_SIZE_LIMIT = 100000;
```

Comment:
Is POW_INT_CACHE_SIZE_LIMIT really a hard coded value?
It might be nice to have a way to change it?